HomeWork2/Clean Architecture/Catalog Service/Infrastructure/Services/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Application.Interfaces.Repositories;
+using Infrastructure.Interfaces.Repositories;
+using Application.Interfaces.Services;
+
+
+namespace Infrastructure.Services;
+
+public static class IServiceCollectionExtensions
+{
+    public static void AddDbContext(this IServiceCollection services, IConfiguration configuration)
+    {
+        var connectionString = configuration.GetConnectionString("DefaultConnection");
+
+         services.AddDbContext<ApplicationDbContext>(
```

Comment:
In clean/onion architecture you have to separate an abstraction (e..g Repository) and its implementation e.g. EF+MS SQL. So you have to move EF and MS SQL specific code to a separate library to be able to switch them like:
services.AddInfrastructureWithEFMsSQL() vs services.AddInfrastructureWithPostGresDapper().

but main logic should rely on generic interface you have already in place

Reply:
updated